product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/src/test/java/org.wso2.identity.scenarios.test.scim2/ManageRolesWithUsersSCIM2TestCase.java

```diff
@@ -0,0 +1,142 @@
+
+        this.username = username;
+        this.password = password;
+        this.tenantDomain = tenantDomain;
+        this.userInputFileName = userInputFileName;
+        this.groupInputFileName = groupInputFileName;
+    }
+
+    @DataProvider(name = "manageRolesConfigProvider")
+    private static Object[][] manageRolesConfigProvider() {
+
+        return new Object[][] {
+                {
+                        ADMIN_USERNAME, ADMIN_PASSWORD, SUPER_TENANT_DOMAIN, "scim2User.json", "scim2Group1.json"
+                }
+        };
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws Exception {
+
+        super.init();
+        scim2Client = new SCIM2CommonClient(getDeploymentProperty(IS_HTTPS_URL));
+        client = HttpClients.createDefault();
+    }
+
+    @Test
+    public void testProvisionUser() throws Exception {
+
+        JSONObject userJSON = scim2Client.getUserJSON(userInputFileName);
+        HttpResponse response = scim2Client.provisionUser(client, userJSON, ADMIN_USERNAME, ADMIN_PASSWORD);
+        assertEquals(response.getStatusLine().getStatusCode(), HttpStatus.SC_CREATED,
+                "User has not been created successfully");
+        JSONObject returnedUserJSON = getJSONFromResponse(response);
+        String userId = returnedUserJSON.get(ID_ATTRIBUTE).toString();
+        users.put(userJSON.get("userName").toString(), userId);
+        assertNotNull(userId, "SCIM2 user id not available in the response.");
+        EntityUtils.consume(response.getEntity());
+    }
+
+    @Test(dependsOnMethods = "testProvisionUser")
+    public void testCreateGroupWithMembers() throws Exception {
+        // Build Group object with members.
+        JSONObject groupJSON = scim2Client.getRoleJSON(groupInputFileName);
+        JSONArray members = new JSONArray();
+        for (Map.Entry<String, String> entry : users.entrySet()) {
+            JSONObject member = new JSONObject();
+            member.put(DISPLAY_ATTRIBUTE, entry.getKey());
+            member.put(VALUE_PARAM, entry.getValue());
+            members.add(member);
+        }
+        groupJSON.put(MEMBERS_ATTRIBUTE, members);
+
+        HttpResponse response = scim2Client.provisionGroup(client, groupJSON, username, password);
+        assertEquals(response.getStatusLine().getStatusCode(), HttpStatus.SC_CREATED,
+                "Group has not been created with members");
+        JSONObject returnedGroupJSON = getJSONFromResponse(response);
+        groupId = returnedGroupJSON.get(ID_ATTRIBUTE).toString();
+        assertNotNull(groupId, "SCIM2 group id not available in the response.");
+        EntityUtils.consume(response.getEntity());
+    }
+
+    @Test(dependsOnMethods = "testCreateGroupWithMembers")
+    private void testDeleteUser() throws Exception {
+
+        for (Map.Entry<String, String> user : users.entrySet()) {
+            HttpResponse response = scim2Client.deleteUser(client, user.getValue(), ADMIN_USERNAME, ADMIN_PASSWORD);
```

Comment:
Isn't this line exceeded 120 characters?

Reply:
No..